apollo-backend/src/main/java/io/logz/apollo/services/SlaveService.java

```diff
@@ -0,0 +1,155 @@
+package io.logz.apollo.services;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Splitter;
+import io.logz.apollo.configuration.ApolloConfiguration;
+import io.logz.apollo.dao.EnvironmentDao;
+import io.logz.apollo.dao.SlaveDao;
+import io.logz.apollo.models.Environment;
+import io.logz.apollo.models.Slave;
+import org.apache.commons.lang3.concurrent.BasicThreadFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.requireNonNull;
+import static org.hibernate.internal.util.collections.CollectionHelper.isEmpty;
+
+@Singleton
+public class SlaveService {
+    private static final Logger logger = LoggerFactory.getLogger(SlaveService.class);
+
+    private final String slaveId;
+    private final ScheduledExecutorService keepaliveExecutorService;
+    private final Boolean isSlave;
+    private final List<Integer> environmentIds;
+    private final ApolloConfiguration apolloConfiguration;
+    private final SlaveDao slaveDao;
+    private final EnvironmentDao environmentDao;
+
+    private boolean started = false;
+
+    @Inject
+    public SlaveService(ApolloConfiguration apolloConfiguration, SlaveDao slaveDao, EnvironmentDao environmentDao) {
+        this.slaveDao = requireNonNull(slaveDao);
+        this.environmentDao = requireNonNull(environmentDao);
+        this.apolloConfiguration = requireNonNull(apolloConfiguration);
+
+        isSlave = apolloConfiguration.getSlave().isSlave();
+        environmentIds = parseEnvironmentIds();
+
+        if (isSlave && isEmpty(environmentIds)) {
+            logger.error("Slave must be bundled with a valid list of environments! Bailing..");
+            throw new RuntimeException("Could not understand slaves params");
+        }
+
+        slaveId = UUID.randomUUID().toString();
```

Comment:
Prefer the usage of the hostname over this, add a variable in the `SlaveConfiguration` named `slaveId` which is nullable, if not exist generate random one like you're doing here.

Reply:
Why does it matter?

Reply:
Visibility we'll be able to see in the DB the environments each  host claimed